src/irr/asset/CShaderIntrospector.cpp

```diff
@@ -62,19 +62,19 @@ E_FORMAT spvImageFormat2E_FORMAT(spv::ImageFormat _imgfmt)
 }
 }//anonymous ns
 
-const SIntrospectionData* CShaderIntrospector::introspect(const ICPUShader* _shader)
+const CIntrospectionData* CShaderIntrospector::introspect(const ICPUShader* _shader)
 {
     if (!_shader)
         return nullptr;
 
     auto found = m_introspectionCache.find(_shader);
     if (found != m_introspectionCache.end())
-        return &found->second; //TODO i think it may break if user will hold this ptr, and run more introspections with this CShaderIntrospector object
+        return found->second.get();
 
     auto introspectSPV = [this](const ICPUShader* _spvshader) {
         const ICPUBuffer* spv = _spvshader->getSPVorGLSL();
         spirv_cross::Compiler comp(reinterpret_cast<const uint32_t*>(spv->getPointer()), spv->getSize()/4u);
-        return doIntrospection(comp, m_entryPoint);
+        return core::smart_refctd_ptr<CIntrospectionData>(doIntrospection(comp, m_entryPoint), core::dont_grab);
```

Comment:
why isn't `doIntrospection` returning a `core::smart_refctd_ptr<CIntrospectionData>`  right away?

Reply:
because it doesn't need to.. no greater purpose really

Reply:
preventing memleaks?

Reply:
resolved